File: imei/imei.py
#salt = "5e8dd316726b0335"      # sim:          hwe620datacard
#salt = "97b7bc6be525ab44"      # flash:        e630upgrade
import hashlib
karimm = "5e8dd316726b0335"
karimmm = "97b7bc6be525ab44"
def Amoug(ay,karim):
  digest = hashlib.md5((ay+karim).lower()).digest()
  code = 0
  for i in range(0,4):
    code += (ord(digest[i])^ord(digest[4+i])^ord(digest[8+i])^ord(digest[12+i])) << (3-i)*8
    code &= 0x1ffffff
    code |= 0x2000000
  return code
ay = raw_input("Insert UR IMEI Number Please: ")
if (ay == ""):
  print "Please Insert UR IMEI Number"
else:
  print "UR Modem Unlock code : %s " % Amoug(ay,karimm);
  print "UR Modem Flash code : %s " % Amoug(ay,karimmm);

File: imei/imei.php
 
<?php
/**
* Huawei E303 | E369 | E357 Free Unlocker
* _Oami_Symbianize_
*
**/
class hw {
	public function calculate($imei, $mode){
			$arrayofbytes = array();
			$digesthash = md5($imei.$this->mode($mode));
			$arrayofbytes = $this->bytearray($digesthash);
			return $this->xorbytes($arrayofbytes);
	}
	private function mode($arg){
			$this->unlock = "5e8dd316726b0335";
			$this->flash = "97b7bc6be525ab44";
			if($arg == 'unlock'){
				return $this->unlock;
			}
			else{
				return $this->flash;
			}
	}
	private function bytearray($hash){
		$splitdigest = substr(chunk_split($hash,2,":"),0,-1);
		$arrdigest = explode(":",$splitdigest);
		return $arrdigest;
	}
	private function xorbytes($arr){
		foreach (range(0,3) as $i) {
			$code = dechex(hexdec($arr[$i]) ^ hexdec($arr[4+$i]) ^ hexdec($arr[8+$i])  ^ hexdec($arr[12+$i]));
			if(strlen($code)< 2) {
				$code = "0" . $code;
			}
			$codes = $codes . $code;
		}
		$tmpcdec = hexdec($codes);
		$tmp1dec = hexdec("1ffffff");
		$tmp2dec = hexdec("2000000");
		$c = $tmpcdec & $tmp1dec;
		$c = $c | $tmp2dec;
		return $c;
	}
}
if(isset($_POST['imei'])){
	$imei = htmlspecialchars($_POST['imei']);
	$hw = new hw();
	if(strlen($imei) == 15){
		$imei = htmlspecialchars($imei);
		echo "<em>Results:</em>";
		echo "<em>Unlock  </em>". $hw->calculate($imei,"unlock");
		echo "<em>Flash  </em>". $hw->calculate($imei,"flash");
		echo "<p/>";
	}
	else{
		if($imei < 15) {
			echo "<em>Your box is empty</em>";
		}
		else{
			echo "<em>Error : Invalid IMEI</em>";
		}
	}
}else{
	echo"Enter your IMEI Number";
}
?>